ckanext/ontario_theme/templates/internal/package/resource_validation.html

```diff
@@ -23,19 +28,75 @@
     </div>
     {% snippet "package/snippets/successful_page_alert.html", success_message=success_message %}
   {% else %}
+    {% set is_csv = "False" %}
     {% set success_message = "Data uploaded successfully" %}
     {% snippet "package/snippets/successful_page_alert.html", success_message=success_message %}
+    <form "method=post" action="{{ action }}">
+      <button id="nextBtn"
+              class="ontario-button ontario-button--primary"
+              name="save"
+              type="submit">{{ _('Continue') }}</button>
+    </form>
   {% endif %}
 
-  {% set package_id = resource.get('package_id') %}
-  {% set resource_id = resource.get('id') %}
-  {% set action = h.url_for('datastore.dictionary',
-      id=package_id,
-      resource_id=resource.id) %}
- 
-  <form method="get" action="{{ action }}">
-    <button class="ontario-button ontario-button--primary"
-            name="save"
-            type="submit">Continue</button>
-  </form>
+  {% if is_csv == "True" %}
+    <script>
+      function poll_datastore() {
+        const resId = "{{ resource.id }}";
+        const xhr = new XMLHttpRequest();
+        const url = `/api/3/action/resource_show?id=${resId}`
+
+        xhr.open("POST", url, true);
+        xhr.setRequestHeader("Content-Type", "application/x-www-form-urlencoded");
+        xhr.send();
+
+        xhr.onreadystatechange = () => {                                
+          if (xhr.readyState === XMLHttpRequest.DONE && xhr.status === 200) {
+            let thisResponse = JSON.parse(xhr.response);
+
+            if (thisResponse.result.datastore_active) {
+              const sleep = (delay) => new Promise((resolve) => setTimeout(resolve, delay))
+         
+              const fixedDelay = async () => {
+                await sleep(2000)
+                document.getElementById("pendingTxt").innerHTML = "Data file uploaded. Please click Continue.";
+                document.getElementById("spinnerSVG").style.display = "none";
+                document.getElementById("step2Btn").style.display = "block";
+              }
+              fixedDelay();
+
+            } else {
+            setTimeout(poll_datastore, 1000);
+            }
+          }
+        }
+      };
+      poll_datastore();
```

Comment:
I don't see any particular reason why this needs to be in the HTML rather than it's own js file (ideally this would be a [ckan js module](https://docs.ckan.org/en/2.9/theming/javascript.html#javascript-modules)). We don't typically use HTML in-file javascript. @aimalkhan What do you think, should it be kept in the HTML or moved to a js file in the scripts directory?

Reply:
The js module are attached to HTML elements in the page. Here we want to call a polling function on page load and not associate with any particular HTML element (until the result of the poll).

Reply:
Okay, if not a js module, then a regular js file will suffice

Reply:
It does not seem possible to pass arguments to JS functions from the jinja template other than through an html element. There are some instances of `<script>` tags in core ckan templates.